### PR TITLE
30 import csv

### PR DESCRIPTION
**Pull request by cprghunter**
corresponding front-end pr: https://github.com/CSC-307-My-Contacts/my-contacts-frontend/pull/67

### CODE REVIEW

File: backend/contactsAPI.py [hunk 1]
```diff
@@ -73,14 +73,110 @@ def create_user():
...
+    return new_db_contacts
+
+def parse_upload_csv(token, fields, contacts):
+    contacts = [[entry.strip() for entry in contact.decode().split(',')] for contact in contacts]
+    user = User().find_by_token(token)
+    if fields[0] == "Name":
+        new_db_contacts = parse_google_csv(fields, contacts)
+    elif fields[0] == "First Name":
+        new_db_contacts = parse_outlook_csv(fields, contacts)
+    else:
```
Comment (baileywickham) [inline backend/contactsAPI.py:169]: Should this be returned as empty?

File: backend/contactsAPI.py [hunk 2]
```diff
@@ -73,14 +73,110 @@ def create_user():
...
+
+        #csv_util_clear_empty_fields(new_db_contact)
+        new_db_contacts.append(new_db_contact)
+    
+    return new_db_contacts
+
+def parse_upload_csv(token, fields, contacts):
+    contacts = [[entry.strip() for entry in contact.decode().split(',')] for contact in contacts]
+    user = User().find_by_token(token)
+    if fields[0] == "Name":
```
Comment (baileywickham) [inline backend/contactsAPI.py:165]: Is there any other way to determine which contacts csv we are getting?

**Review by AlexMaclean — CHANGES_REQUESTED** [review]
So far looks really good, I was able to import *nearly* my entire contact list without issue.

The first major bug I've identified while testing is that your code doesn't handle quote wrapped CSV cells with commas inside of them. I have a contact with the name `Lisa, Joe's mom` when I try to import her I see the name as `"Lisa` and the rest of the fields get lost. I suspect there are other weird edge cases like this in csv format. Looking into a python CSV library might be a good idea.

File: backend/contactsAPI.py (hunk 2 above)
Comment (cprghunter) [reply backend/contactsAPI.py:165]: This is definitely the easiest method I found, and always works to separate the two.

File: backend/contactsAPI.py (hunk 1 above)
Comment (cprghunter) [reply backend/contactsAPI.py:169]: Resolved based on discussion in meeting.

**Comment by cprghunter** [comment]
> So far looks really good, I was able to import _nearly_ my entire contact list without issue.
> 
> The first major bug I've identified while testing is that your code doesn't handle quote wrapped CSV cells with commas inside of them. I have a contact with the name `Lisa, Joe's mom` when I try to import her I see the name as `"Lisa` and the rest of the fields get lost. I suspect there are other weird edge cases like this in csv format. Looking into a python CSV library might be a good idea.

Most recent commit makes these changes.

**Comment by baileywickham** [comment]
Looks good to me!

**Review by AlexMaclean — APPROVED** [review]
After these changes it looks good to me too, Thanks!

**Pull request merged**